nntrainer/opencl/opencl_context_manager.cpp

```diff
@@ -155,6 +155,31 @@ bool ContextManager::CreateDefaultGPUDevice() {
   device_id_ = devices[0];
   platform_id_ = platform_id_;
 
+#ifdef ENABLE_FP16
+  // check for fp16 (half) support available on device
```

Comment:
which half-precision does it support? `_Float16`, `__fp16`, or `half`

Reply:
Ideally all of `_Float16`, `__fp16` and `half` can be used interchangeably inside kernels when `cl_khr_fp16` extension is enabled. I have tested with `__fp16` and `half` myself, they worked exactly the same. However I did not test with `__Float16` but it is supported by OpenCL 2.1 or later.

Reply:
`half` is specifically designed for use in OpenCL kernels. Some are architecture specific  (e.g: `__fp16` for ARM). So `half` will be more portable and benefit from further optimizations..

Reply:
`half` will only work inside OpenCL kernels. So I think adding it to tensor data type won't be useful in this scenario.